pkg/services/accesscontrol/ossaccesscontrol/permissions_services.go

```diff
@@ -170,14 +170,17 @@ func ProvideDashboardPermissions(
 			return nil
 		},
 		InheritedScopesSolver: func(ctx context.Context, orgID int64, resourceID string) ([]string, error) {
+			wildcards := accesscontrol.WildcardsFromPrefix(dashboards.ScopeFoldersPrefix)
+			scopes := []string(wildcards)
+
 			dashboard, err := getDashboard(ctx, orgID, resourceID)
 			if err != nil {
 				return nil, err
 			}
 			metrics.MFolderIDsServiceCount.WithLabelValues(metrics.AccessControl).Inc()
 			// nolint:staticcheck
-			if dashboard.FolderID > 0 {
-				query := &dashboards.GetDashboardQuery{ID: dashboard.FolderID, OrgID: orgID}
+			if dashboard.FolderUID != "" {
+				query := &dashboards.GetDashboardQuery{UID: dashboard.FolderUID, OrgID: orgID}
```

Comment:
Note: this change is unrelated to the bug - I just updated the code to use folder UIDs not IDs (as IDs are deprecated).